### PR TITLE
dev-java/gradle: bump to 3.0

### PR DESCRIPTION
**Pull request by msva**

### CODE REVIEW

File: dev-java/gradle/gradle-3.0.ebuild [hunk 1]
```diff
@@ -11,34 +10,44 @@ SRC_URI="http://services.gradle.org/distributions/${P}-src.zip"
...
 	# because then gradle won't find them.
-	cd lib || die "lib/ not found"
+	cd lib;
 	for jar in *.jar; do
 		java-pkg_newjar ${jar} ${jar}
 	done
 
 	# plugins in lib/plugins
-	cd plugins
+	cd plugins;
```
Comment (chewi) [inline dev-java/gradle/gradle-3.0.ebuild]: Why?

File: dev-java/gradle/gradle-3.0.ebuild [hunk 2]
```diff
@@ -11,34 +10,44 @@ SRC_URI="http://services.gradle.org/distributions/${P}-src.zip"
...
+	cd dist;
+	dodoc changelog.txt getting-started.html
+
 	insinto "${gradle_dir}"
 
 	# jars in lib/
 	# Note that we can't strip the version from the gradle jars,
 	# because then gradle won't find them.
-	cd lib || die "lib/ not found"
+	cd lib;
```
Comment (chewi) [inline dev-java/gradle/gradle-3.0.ebuild]: The message isn't necessary but you should always `|| die` on a `cd` call.

File: dev-java/gradle/gradle-3.0.ebuild [hunk 3]
```diff
@@ -11,34 +10,44 @@ SRC_URI="http://services.gradle.org/distributions/${P}-src.zip"
...
+
+src_prepare() {
+	default
+	java-pkg-2_src_prepare
+}
 
 src_compile() {
-	gradle --gradle-user-home "$WORKDIR" install -Pgradle_installPath=dist || die
+	local inst_target="install"
+	use doc && inst_target="installAll"
```
Comment (chewi) [inline dev-java/gradle/gradle-3.0.ebuild]: How about…?

```sh
local inst_target=$(usex doc installAll install)
```

File: dev-java/gradle/gradle-3.0.ebuild [hunk 4]
```diff
@@ -11,34 +10,44 @@ SRC_URI="http://services.gradle.org/distributions/${P}-src.zip"
 HOMEPAGE="http://www.gradle.org/"
 LICENSE="Apache-2.0"
 SLOT="${PV}"
-KEYWORDS="~amd64 ~x86"
+KEYWORDS="~x86 ~amd64"
```
Comment (chewi) [inline dev-java/gradle/gradle-3.0.ebuild]: If they'd been in the wrong order to start with, I'd have let you off but what was the point of this? I know this is being picky but other devs will pick you up on this too. Let's keep the noise down. :smiley_cat:

**Comment by chewi** [comment]
I don't know anything about Gradle but I'm guessing it's not something we need to SLOT?

**Comment by msva** [comment]
> I don't know anything about Gradle but I'm guessing it's not something we need to SLOT?
Actually, I thought about that too. But, on the other hands, theoretically, it can be the situation, when some old package can require some pretty old gradle version.
// although, actually, many packages bundles gradle anyway :-/

All other things is fixed.

**Comment by chewi** [comment]
Merged, thanks. I am assuming the change from `gradle` to `./gradlew` is correct because I can't see how it could have worked before if you didn't already have gradle installed.

**Comment by chewi** [comment]
By the way, I'm really not impressed by how it downloads something during the build, there's no way I'd let that into the tree. I merged it anyway because the previous version did this too but if you could find a way to avoid that, it'd be great. Surely it must be possible to add that prebuilt zip to SRC_URI.

**Comment by msva** [comment]
Yeah, I dislike downloading anything during something that is after src_unpack too. But unfortunatelly, I'm not that gradle guru to know how to disallow it from doing that. Although, I think it is possible with a bunch of patches against buildsystem...

**Pull request closed without merging**